volatility3/framework/plugins/linux/psscan.py

```diff
@@ -0,0 +1,127 @@
+# This file is Copyright 2021 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+from typing import Iterable, List, Tuple
+import struct
+
+from volatility3.framework import renderers, interfaces, symbols, constants
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+from volatility3.framework.layers import scanners
+from volatility3.framework.renderers import format_hints
+
+
+class PsScan(interfaces.plugins.PluginInterface):
+    """Scans for processes present in a particular linux image."""
+
+    _required_framework_version = (2, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+        ]
+
+    def _get_task_fields(
+        self, task: interfaces.objects.ObjectInterface
+    ) -> Tuple[int, int, int, str]:
+        """Extract the fields needed for the final output
+
+        Args:
+            task: A task object from where to get the fields.
+        Returns:
+            A tuple with the fields to show in the plugin output.
+        """
+        pid = task.tgid
+        tid = task.pid
+        ppid = task.parent.tgid if task.parent else 0
+        name = utility.array_to_string(task.comm)
+
+        task_fields = (format_hints.Hex(task.vol.offset), pid, tid, ppid, name)
+        return task_fields
+
+    def _generator(self):
+        """Generates the tasks found from scanning."""
+
+        for task in self.scan_tasks(
+            self.context, self.config["kernel"], self.config["kernel.symbol_table_name"]
+        ):
+            row = self._get_task_fields(task)
+            yield (0, row)
+
+    @classmethod
+    def scan_tasks(
+        cls,
+        context: interfaces.context.ContextInterface,
+        vmlinux_module_name: str,
+        kernel_table_name: str
+    ) -> Iterable[interfaces.objects.ObjectInterface]:
+        """Scans for tasks in the memory layer.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            vmlinux_module_name: The name of the kernel module on which to operate
+        Yields:
+            Task objects
+        """
+        vmlinux = context.modules[vmlinux_module_name]
+
+        # check if this image is 32bit or 64bit
+        is_32bit = not symbols.symbol_table_is_64bit(
+            context, vmlinux.symbol_table_name
+        )
+        if is_32bit:
+            pack_format = "I"
+            kernel_add = 0
+        else:
+            pack_format = "Q"
+            kernel_add = 0xffff000000000000
```

Comment:
This wont work as I intended all the time for the reasons listed in this issue (PML4 v LA57) - It should also probably be an 'or' rather than an 'add' -  https://github.com/volatilityfoundation/volatility3/issues/883

Reply:
#702 at your service.